src/util/org-utils.ts

```diff
@@ -0,0 +1,195 @@
+                this.EntityListFileName
+            );
+            if (!fs.existsSync(entityListFile)) {
+                const objectList = await this.getEntityList(this.connection);
+                this.entities = objectList;
+                fs.writeFileSync(entityListFile, JSON.stringify(objectList), {
+                    mode: 0o666
+                });
+            } else {
+                const entityContent = fs.readFileSync(entityListFile, 'utf8');
+                this.entities = JSON.parse(entityContent);
+            }
+            this.isUserLoggedIn = true;
+
+            return connect;
+        } catch (error) {
+            this.isUserLoggedIn = false;
+            this.connection = undefined;
+            return undefined;
+        }
+    }
+
+    public static async getEntityList(connection: Connection): Promise<string[]> {
+        const globalResult = await connection.describeGlobal();
+        return globalResult.sobjects.map((sobjettResult) => sobjettResult.name);
+    }
+
+    // Retrieves objectInfo folder path, which is '<projectRoot>/.sf/orgName/objectInfos/'
+    public static objectInfoFolderPath(): string {
+        const orgPath = this.ORG_NAME === undefined ? './' : this.ORG_NAME;
+        const projectPath = path.resolve(__dirname, '../../../../../');
+        const objectInfoFolder = path.posix.join(
+            projectPath,
+            OrgUtils.SF_FOLDER,
+            orgPath,
+            OrgUtils.OBJECT_INFO_FOLDER
+        );
+        if (!fs.existsSync(objectInfoFolder)) {
+            fs.mkdirSync(objectInfoFolder, { recursive: true });
+        }
+        return objectInfoFolder;
+    }
+
+    public static fetchObjectInfoFromDisk(objectApiName: string): string | undefined {
+        const objectInfoJsonFile = path.posix.join(
+            this.objectInfoFolderPath(),
+            `${objectApiName}.json`
+        );
+        if (!fs.existsSync(objectInfoJsonFile)) {
+            return undefined;
+        }
+
+        return fs.readFileSync(objectInfoJsonFile, 'utf-8');
+    }
+
+    // Acquires ObjectInfo data by first searching in memory, then on disk, and finally over the network.
+    public static getObjectInfo(objectApiName: string): string | undefined {
+        OrgUtils.checkAuthStatus();
+        if (!OrgUtils.isUserLoggedIn) {
+            this.objectInfoCache.clear();
+            return undefined;
+        }
+        // Checks mem cache
+        let objectInfo = this.objectInfoCache.get(objectApiName);
+
+        if (objectInfo !== undefined) {
+            return objectInfo;
+        }
+
+        // Checks disk cache
+        objectInfo = this.fetchObjectInfoFromDisk(objectApiName);
+        if (objectInfo !== undefined) {
+            this.objectInfoCache.set(objectApiName, objectInfo);
+            return objectInfo;
+        }
+
+        // Network loading is going on
+        if (this.objectInfoLoading.has(objectApiName)) {
+            return undefined;
```

Comment:
We shouldn't turn away concurrent requests just because they haven't been fulfilled over the network yet. I would propose that, given that this method needs to be converted to an async method anyway, we should establish a model where the network request can be made in the form of a singular `Promise`, that would then fulfill multiple requests.  Something like this model:

```typescript
let objectInfoResponsePromise: Promise<SomeObjectInfoType> | null = null;

public static async getObjectInfo(objectApiName: string): Promise<SomeObjectInfoType> {
    // Any nominal case return values here, e.g. value's already in the cache, etc.
    return alreadyEstablishedObjectInfoData;

    // Otherwise, make a network call.
    if (!objectInfoResponsePromise) {
        objectInfoResponsePromise = new Promise<SomeObjectInfoType>((resolve, reject) => {
            // Do the network stuff to get the ObjectInfo data back.
            const objectInfoResponse = theResponse;
            resolve(objectInfoResponse);
        });

        // Once the network request is complete, reset the promise
        objectInfoResponsePromise.finally(() => {
            objectInfoResponsePromise = null;
        });
    }

    return objectInfoResponsePromise;
}
```

That way, _anyone_ who calls the method will get a value back, even if that network call is in flight. It just creates a "singleton" network promise that will fulfill for everyone who calls the method.

Reply:
This is nice!

Reply:
Will update it into async method